### PR TITLE
[MM-41851] - Add Start Trial to admin onboarding checklist

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Adds a check list to prompt starting trial

#### Ticket Link
[MM-41851](https://mattermost.atlassian.net/browse/MM-41851)

#### Screenshots
Screenshot 2022-02-17 at 16 34 19

#### Release Note

```release-note
NONE

```

### CODE REVIEW

File: components/global_onboarding_list/onboarding_checklist.tsx [hunk 1]
```diff
@@ -196,6 +196,11 @@ const TaskList = (): JSX.Element => {
             id='onboarding_checklist.task_system_console'
             defaultMessage='Visit the System Console to configure your workspace'
         />,
+        <FormattedMessage
+            key={7}
+            id='onboarding_checklist.task_start_trial'
+            defaultMessage='Learn more about Enterprise-level high-security features'
+        />,
```
Comment (marianunez) [inline components/global_onboarding_list/onboarding_checklist.tsx:203]: We should only show is they are not in cloud, haven't done a trial before and don't have a license (which most likely hasn't happened early in the onboarding but we should be consistent with the same checks we currently do to show this in the main menu)

**Review by marianunez — CHANGES_REQUESTED** [review]

File: i18n/en.json [hunk 2]
```diff
@@ -3845,6 +3845,7 @@
   "onboarding_checklist.task_invite": "Invite team members to the workspace",
   "onboarding_checklist.task_system_console": "Visit the System Console to configure your workspace",
   "onboarding_checklist.video_title": "Watch overview",
+  "onboarding_checklist.task_start_trial": "Learn more about Enterprise-level high-security features",
```
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply i18n/en.json:3848]: Hey @julmondragon yes that's true. I hadn't run `make i18n-extract` yet

File: components/onboarding_tasks/onboarding_tasks_manager.tsx [hunk 3]
```diff
@@ -77,17 +82,34 @@ const taskLabels = {
...
+    // Show this CTA if the instance is currently not licensed and has never had a trial license loaded before
+    const showStartTrialTask = (isCurrentLicensed === 'false' && isPrevLicensed === 'false');
     const list: Record<string, string> = {...OnboardingTasksName};
     if (!pluginsList.focalboard) {
         delete list.BOARDS_TOUR;
     }
     if (!pluginsList.playbooks) {
         delete list.PLAYBOOKS_TOUR;
     }
+    if (!showStartTrialTask) {
```
Comment (neallred) [inline components/onboarding_tasks/onboarding_tasks_manager.tsx:109]: :trophy: Nice checks!!

File: components/onboarding_tasks/onboarding_tasks_manager.tsx [hunk 4]
```diff
@@ -77,17 +82,34 @@ const taskLabels = {
             defaultMessage='Visit the System Console to configure your workspace'
         />
     ),
+    [OnboardingTasksName.START_TRIAL]: (
+        <FormattedMessage
+            id='onboarding_checklist.task_start_trial'
+            defaultMessage='Learn more about Enterprise-level high-security features'
```
Comment (julmondragon) [inline components/onboarding_tasks/onboarding_tasks_manager.tsx:88]: It looks like the message id's for the tasks were update to the format `onboardingTask.checklist.task_...`

**Review by julmondragon — APPROVED** [review]

**Review by neallred — APPROVED** [review]

File: components/onboarding_tasks/onboarding_tasks_manager.tsx [hunk 5]
```diff
@@ -6,20 +6,23 @@ import {FormattedMessage} from 'react-intl';
 import {useDispatch, useSelector} from 'react-redux';
 import {matchPath, useLocation} from 'react-router-dom';
 
-import {trackEvent as trackEventAction} from 'actions/telemetry_actions';
+import {trackEvent, trackEvent as trackEventAction} from 'actions/telemetry_actions';
```
Comment (marianunez) [inline components/onboarding_tasks/onboarding_tasks_manager.tsx:9]: Is this defined twice?

**Review by marianunez — APPROVED** [review]
Are there any tests we can add to cover this new case?

File: components/onboarding_tasks/onboarding_tasks_manager.tsx (hunk 5 above)
Comment (neallred) [reply components/onboarding_tasks/onboarding_tasks_manager.tsx:9]: That's a good catch. In the original code, there was a function argument in this file named `trackEvent` separate from the function call being imported here. For consistency, we might want to stick with `trackEventAction` for the call this PR adds, too.

**Review by stevemudie — APPROVED** [review]
Start Trial appeared as the final items on the checklist so that part is good.

However, The first item on the list (Channels) didn't show as completed until I followed through with every step of the onboarding tour if that's of any concern.

**Pull request merged**